pop/service/service.go

```diff
@@ -98,7 +100,11 @@ type Service struct {
 	// Sync tools
 	// key of map is ID of party
 	// synchronizing inside one party
-	syncs map[string]*sync
+	syncs map[string]*syncChans
+	// verifyFinalBuffer is a temporary buffer for bftVerifyFinal results
+	verifyFinalBuffer sync.Map
```

Comment:
This needs way more documentation. Why do we need these sync.Maps? Who are the producers, consumers? How do we know that they are cleaned up?